core/src/Platforms/iOS/TokenCacheAccessor.cs

```diff
@@ -32,90 +32,248 @@
 using Microsoft.Identity.Core;
 using Microsoft.Identity.Core.Cache;
 using System.Collections.ObjectModel;
+using Microsoft.Identity.Core.Helpers;
 
 namespace Microsoft.Identity.Core
 {
     internal class TokenCacheAccessor : ITokenCacheAccessor
     {
-        private const string AccessTokenServiceId = "com.microsoft.identity.client.accessToken";
-        private const string RefreshTokenServiceId = "com.microsoft.identity.client.refreshToken";
-        private const string IdTokenServiceId = "com.microsoft.identity.client.idToken";
-        private const string AccountServiceId = "com.microsoft.identity.client.Account";
+        public const string CacheKeyDelimiter = "-";
+
+        static Dictionary<string, int> AuthorityTypeToAttrType = new Dictionary<string, int>()
+        {
+            {AuthorityType.AAD.ToString(), 1001},
+            {AuthorityType.MSA.ToString(), 1002},
+            {AuthorityType.MSSTS.ToString(), 1003},
+            {AuthorityType.OTHER.ToString(), 1004},
+        };
+
+        enum CredentialAttrType
+        {
+            AccessToken = 2001,
+            RefreshToken = 2002,
+            IdToken = 2003,
+            Password = 2004
+        }
 
         private const bool _defaultSyncSetting = false;
         private const SecAccessible _defaultAccessiblityPolicy = SecAccessible.AfterFirstUnlockThisDeviceOnly;
 
+        private readonly string DefaultKeychainGroup = "com.microsoft.adalcache";
+        private readonly string TeamIdKey = "teamIDHint";
+
+        private string keychainGroup;
+
+        private string GetBundleId()
+        {
+            return NSBundle.MainBundle.BundleIdentifier;
+        }
+   
+        public void SetSecurityGroup(string securityGroup)
+        {
+            if (securityGroup == null)
+            {
+                keychainGroup = GetBundleId();
+            }
+            else
+            {
+                keychainGroup = securityGroup;
+            }
+        }
+
+        private string GetTeamId()
+        {
+            var queryRecord = new SecRecord(SecKind.GenericPassword)
+            {
+                Service = "",
+                Account = TeamIdKey
+            };
+
+            SecRecord match = SecKeyChain.QueryAsRecord(queryRecord, out SecStatusCode resultCode);
+
+            if (resultCode == SecStatusCode.ItemNotFound)
+            {
+                SecKeyChain.Add(queryRecord);
+                match = SecKeyChain.QueryAsRecord(queryRecord, out resultCode);
+            }
+
+            return match.AccessGroup.Split('.')[0];
```

Comment:
does this always work, i.e. returns a string with a dot inside?

Reply:
yes access group in iOS has particular format